### PR TITLE
Fix bug in GlueJobOperator where consecutive runs fail when a local script file is used

### LINKED ISSUE #38959: GlueJobOperator with local script location fails on consecutive runs
### Apache Airflow Provider(s)

amazon

### Versions of Apache Airflow Providers

[apache-airflow-providers-amazon](https://airflow.apache.org/docs/apache-airflow-providers-amazon/8.19.0)	8.19.0

### Apache Airflow version

2.8.3

### Operating System

Amazon Linux 2; Kernel Version: 5.10.209-198.812.amzn2.x86_64

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

We deploy airflow on EKS using the official Helm chart.

### What happened

We are deploying a Glue Job using the GlueJobOperator with the following configuration:
```
GlueJobOperator(
        job_name="weather_data_prepared_local_file",
        script_location=str(Path(__file__).resolve().parent / "scripts/weather_data_prepared.py"),
        s3_bucket="aws-glue-temporary-bucket",
        task_id="WeatherGlueJob",
        iam_role_name="eks.data.airflow.glue.executor",
        create_job_kwargs={"GlueVersion": "4.0", "NumberOfWorkers": 2, "WorkerType": "G.1X"},
        update_config=True,
        aws_conn_id='datalake',
        dag=dag
    )
```

This works fine for the first run of our DAG and the script file gets uploaded to `artifacts/glue-scripts/weather_data_prepared.py`
However, when we trigger the DAG for a second run, it fails because the file already exists.

```
 [2024-04-12T08:13:33.615+0000] {glue.py:173} INFO - Initializing AWS Glue Job: weather_data_prepared 
 [2024-04-12T08:13:33.659+0000] {base.py:83} INFO - Using connection ID 'datalake' for task execution
 [2024-04-12T08:13:34.265+0000] {taskinstance.py:2731} ERROR - Task failed with exception Traceback (most recent call last): 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 44 result = _execute_callable(context=context, **execute_callable_kwargs) 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 41 return execute_callable(context=context, **execute_callable_kwargs) 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/g glue_job_run = self.glue_job_hook.initialize_job(self.script_args, self.run_job_kwargs) 
 File "/usr/local/lib/python3.8/functools.py", line 967, in __get__ val = self.func(instance) 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/g s3_hook.load_file( 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/s3.py return func(*bound_args.args, **bound_args.kwargs) 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/s3.py return func(*bound_args.args, **bound_args.kwargs) 
 File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/s3.py raise ValueError(f"The key {key} already exists.") 
 ValueError: The key artifacts/glue-scripts/weather_data_prepared.py already exists.
```

### What you think should happen instead

We are of the opinion, that the file on S3 should be overwritten for subsequent DAG executions.
So that consecutive runs of GlueJobOperators using local script locations do not fail.
This enables us to subject our script files to version control and CI/CD pipelines.

### How to reproduce

    1. Create a DAG with the GlueJobOperator, referencing a local file and a S3 Bucket.
    2. Run the DAG twice.

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by moritzsanne**
closes: #38959

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/glue.py [hunk 1]
```diff
@@ -143,7 +143,10 @@ def glue_job_hook(self) -> GlueJobHook:
             s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
             script_name = os.path.basename(self.script_location)
             s3_hook.load_file(
-                self.script_location, self.s3_artifacts_prefix + script_name, bucket_name=self.s3_bucket
+                self.script_location,
+                self.s3_artifacts_prefix + script_name,
+                bucket_name=self.s3_bucket,
+                replace=True,
```
Comment (eladkal) [inline airflow/providers/amazon/aws/operators/glue.py]: Are you sure that this is right for ALL usages of the operator?
If not then this parameter should be set on the operator level thus users can decide on the behavior they want

Comment (moritzsanne) [reply airflow/providers/amazon/aws/operators/glue.py]: Good point. I do not know that this is the right approach for all potential use cases and I agree that implementing this as a parameter in the operator level is the defensive approach. I've just commited an updated implementation.

**Review by o-nikolas — APPROVED** [review]

**Comment by eladkal** [comment]
can you add unit test to cover this change?

**Review by eladkal — APPROVED** [review]

**Comment by moritzsanne** [comment]
Hey, thanks for reviewing this so quickly. I've rebased my branch using the rebase button in github after your reviews. I am unsure if this was the right thing today in this case. I hope this doesn't cause any confusion in the commit history. Let me know if you need anything else from me to get this merged. 

Enjoy your weekend!

**Pull request merged**